### PR TITLE
Feature/map component

### PR DESCRIPTION
**Pull request by nbumbarger**
Adds functional map to the map placeholder, and connects it to the API to visualize the collection of requests (in the home view), individual requests (in the requests view), a user's requests (in the dashboard view), and individual tasks (in the task view).

### CODE REVIEW

**Comment by nbumbarger** [comment]
This is ready. Most views include the view map, and the task page temporarily contains an example of the edit component.

File: app/assets/scripts/components/display-map.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+import config from '../config';
+
+const DisplayMap = React.createClass({
+  displayName: 'DisplayMap',
+
+  propTypes: {
+    mapId: T.string,
+    results: T.object,
+    mapClass: T.string
```
Comment (danielfdsilva) [inline app/assets/scripts/components/display-map.js]: Better to use `className` so it's a common property.

File: app/assets/scripts/components/display-map.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  },
+
+  componentDidMount: function () {
+    mapboxgl.accessToken = config.mbToken;
```
Comment (danielfdsilva) [inline app/assets/scripts/components/display-map.js]: move to `main.js` so we only do it once across the site. No need for duplication.

File: app/assets/scripts/components/edit-map.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import React, { PropTypes as T } from 'react';
+import mapboxgl from 'mapbox-gl';
+import GLDraw from 'mapbox-gl-draw';
+
+import config from '../config';
+
+import { mbStyles } from '../utils/mapbox-styles';
+
+const EditMap = React.createClass({
+  displayName: 'DisplayMap',
```
Comment (danielfdsilva) [inline app/assets/scripts/components/edit-map.js]: Wrong `displayName`

File: app/assets/scripts/components/edit-map.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+  propTypes: {
+    mapId: T.string,
+    mapClass: T.string
```
Comment (danielfdsilva) [inline app/assets/scripts/components/edit-map.js:13]: Better to use `className` so it's a common property.

File: app/index.html [hunk 5]
```diff
@@ -43,13 +43,16 @@
     <!-- build:css /assets/styles/main.css -->
     <link rel="stylesheet" href="/assets/styles/main.css">
     <!-- endbuild -->
-  
+
+    <link rel="stylesheet" href="/node_modules/mapbox-gl/dist/mapbox-gl.css">
+    <link rel="stylesheet" href="/node_modules/mapbox-gl-draw/dist/mapbox-gl-draw.css">
```
Comment (danielfdsilva) [inline app/index.html]: These need to be included in the build process above (line 43-45).

**Review by danielfdsilva — CHANGES_REQUESTED** [review]
Other notes:
- Please add the token to `production.js` This is not sensitive and unless we go with a different map, won't change. (update the readme to reflect this)

File: README.md [hunk 6]
```diff
@@ -14,6 +14,14 @@ $ npm install
 
 ### Getting started
 
+Before launching, the file at /app/assets/config/local.js must export an object containing the API's address and a valid Mapbox token, e.g.
+```
+module.exports = {
+  api: 'http://localhost:4000',
+  mbToken: 'pk.eyJ1I...'
```
Comment (danielfdsilva) [inline README.md]: Since this is in the `production.js` which is committed, we can remove it from here

File: app/index.html [hunk 7]
```diff
@@ -42,14 +42,16 @@
 
     <!-- build:css /assets/styles/main.css -->
     <link rel="stylesheet" href="/assets/styles/main.css">
+    <link rel="stylesheet" href="/node_modules/mapbox-gl/dist/mapbox-gl.css">
```
Comment (danielfdsilva) [inline app/index.html]: This was actually my bad.
Can we move these 2 to the vendor build? lines 40-41

**Review by danielfdsilva — CHANGES_REQUESTED** [review]

**Review by danielfdsilva — APPROVED** [review]

**Pull request merged**